### ISSUE TITLE
a11yNumResults and a11yNumSelected called too many times

### ISSUE THREAD

**Issue by RichCaloggero** [ericgio/react-bootstrap-typeahead#321]
## a11yNumResults and a11yNumSelected called too many times

The functions a11yNumResults and a11yNumSelected are being called too many times.
From what I can gather from the docs, they should be called when either the result set changes or when the selection set changes, respectively, they they should only be called *once* when the change occurs.

### Steps to reproduce

Using the following typeahead:

- open the page after dropping it into a new react project
- type the letter "c" into the input
- immediately switch to the javascript console and examine the log; you can clearly see the a11yNumResults function being called twice

This produces duplicate screen reader announcements -- very annoying!

### expected behavior

- log should only contain one message
- screen reader should only announce number of matches once each time match set changes

```jsx
<Typeahead
  emptyLabel=""
  //a11yNumSelected={() => ""}
  a11yNumResults={(results) => {
    let n = results.length;
    console.log (`a11yNumResults: ${n}`);
    return `${n.toString()} matches.`;
  }}
  onChange={(selected) => {
    this.setState({selected})
  }}
  options={["chocolate", "vanilla", "strawberry", "coconut"]}
  minLength={1}
  inputProps={{"aria-label": "Choose your favorite flavor:"}}
  placeholder="Choose your favorite flavor:"
/>
```

**Comment by ericgio** [comment]
I see that the initial keypress triggers 2 console logs, but after that it's just 1, as expected. That said, when I turn VoiceOver on, I still only hear 1 announcement.

**Comment by RichCaloggero** [comment]
What browser are you using with VoiceOver?
Chrome works, but is flakey.
Firefox doesn't really work at all.
The only truly usable browser with VoiceOver is Safari!

**Comment by ericgio** [comment]
> The only truly usable browser with VoiceOver is Safari!

The examples seem to work just fine for me with Safari + VoiceOver. No repeat announcements, and #301 is not an issue either. Given that, it's unclear to me whether blame for the bugginess lies with the component or with poor interaction between VoiceOver and browsers other than Safari. It's also unclear to me how other screen readers interact with different browsers and the component, since I currently don't have any way to test other screen readers.

**Comment by ericgio** [comment]
Note that I'm planning on removing `a11yNumResults` and `a11yNumSelected` in the next major release (4.0), since they're not part of the [authoring practices](https://www.w3.org/TR/wai-aria-practices-1.1/#combobox). If this functionality is desired, you'll be able to add it yourself via a [`children` render function](https://github.com/ericgio/react-bootstrap-typeahead/blob/master/src/Typeahead.react.js#L99).
